umake/frameworks/rust.py

```diff
@@ -0,0 +1,144 @@
+        if "{}-unknown-linux-gnu.tar.gz".format(self.arch_trans[arch]) in line:
+            in_download = True
+        if in_download:
+            p = re.search(r'href="(.*)">', line)
+            with suppress(AttributeError):
+                url = p.group(1)
+            p = re.search(r'<td><tt>(\w+)</tt></td>', line)
+            with suppress(AttributeError):
+                sha1 = p.group(1)
+            if "</tr>" in line:
+                in_download = False
+
+        if url is None and sha1 is None:
+            return (None, in_download)
+        return ((url, sha1), in_download)
+
+    @MainLoop.in_mainloop_thread
+    def get_metadata_and_check_license(self, result):
+        """Override this so we can use BS and fetch the checksum separately."""
+        logger.debug("Fetched download page, parsing.")
+
+        page = result[self.download_page]
+
+        error_msg = page.error
+        if error_msg:
+            logger.error("An error occurred while downloading {}: {}".format(self.download_page_url, error_msg))
+            UI.return_main_screen()
+
+        soup = BeautifulSoup(page.buffer)
+
+        link = (soup.find('div', class_="install")
+                .find('td', class_="inst-type", text="Linux (.tar.gz)")
+                .parent
+                .find(text=self.arch_trans[get_current_arch()])
+                .parent
+                .parent)
+
+        if link is None:
+            logger.error("Can't parse the download URL from the download page.")
+            UI.return_main_screen()
+
+        download_url = link.attrs['href']
+        checksum_url = download_url + '.sha256'
+        logger.debug("Found download URL: " + download_url)
+        logger.debug("Downloading checksum first, from " + checksum_url)
+
+        def checksum_downloaded(results):
+            checksum_result = next(iter(results.values()))  # Just get the first.
+            if checksum_result.error:
+                logger.error(checksum_result.error)
+                UI.return_main_screen()
+
+            checksum = checksum_result.buffer.getvalue().decode('utf-8').split()[0]
+            logger.info('Obtained SHA256 checksum: ' + checksum)
+
+            self.download_requests.append(DownloadItem(download_url,
+                                                       checksum=Checksum(ChecksumType.sha256, checksum),
+                                                       ignore_encoding=True))
+            self.start_download_and_install()
+
+        DownloadCenter([DownloadItem(checksum_url)], on_done=checksum_downloaded, download=False)
+
+    def post_install(self):
+        """Add rust necessary env variables"""
+        add_env_to_user(self.name, {"PATH": {"value": os.path.join(self.install_path, "rustc", "bin")},
+                                    "PATH": {"value": os.path.join(self.install_path, "cargo", "bin")},
+                                    "LD_LIBRARY_PATH": {"value": os.path.join(self.install_path, "rustc", "lib")}})
```

Comment:
Please add in the large tests a way a check that the LD_LIBRARY_PATH has been added.